packages/flutter_tools/lib/src/android/android_sdk.dart

```diff
@@ -426,22 +426,6 @@ class AndroidSdk {
       return fileSystem.path.join(javaHomeEnv, 'bin', 'java');
     }
 
```

Comment:
Looks like we have no unit tests for this function. Could you add tests that validate we discover the right path in the situations where:

1. it's present in Android Studio
2. not present in AS but JAVA_HOME is set
3. neither 1 nor 2, but `which java` returns a valid value
4. they really don't have any java binaries we can find

Reply:
done

Reply:
FWIW I think my pr #123916 might have added some tests.